tests/unit/test_application.py

```diff
@@ -236,3 +236,12 @@ def test_parse_info_integrated(monkeypatch, mocker, new_dir):
     assert snap_yaml["summary"] == "Sample summary"
     assert snap_yaml["description"] == "Sample description"
     assert snap_yaml["version"] == "1.2.3"
+
+
+def test_application_plugins():
```

Comment:
Could you add a test verifying if the `list-plugins` command is returning the expected list of plugins for different bases?

Reply:
I can't do that because the `list-plugins` command hasn't been ported to core24 yet (I don't know if it's on our radar)

Reply:
`list-plugins` could be an interesting thing to add to craft-application to get all tools covered.